modules/storages/app/contracts/storages/last_project_folders/base_contract.rb

```diff
@@ -0,0 +1,41 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages::LastProjectFolders
+  class BaseContract < ::ModelContract
+    include ActiveModel::Validations
+
+    attribute :projects_storage
+    validates_presence_of :projects_storage
+    attribute :mode
+    validates :mode, presence: true, inclusion: {
+      in: Storages::ProjectStorage.project_folder_modes.keys.reject { |v| v == :inactive }
```

Comment:
The result of `Storages::ProjectStorage.project_folder_modes.keys.reject { |v| v == :inactive }` is `["inactive", "manual", "automatic"]`. `Storages::ProjectStorage.project_folder_modes` keys are strings. 
```suggestion
      in: Storages::ProjectStorage.project_folder_modes.keys.reject { |v| v == 'inactive' }
```
or just be explicit...
```suggestion
      in: %w[manual automatic]

Reply:
I think it would be nice to have tests for that.

Reply:
replaced with explicit list and added contract unit test